storage/local/storage.go

```diff
@@ -716,20 +697,20 @@ func (s *memorySeriesStorage) preloadChunksForRange(
 	if !ok {
 		has, first, last, err := s.persistence.hasArchivedMetric(fp)
 		if err != nil {
-			return nil, err
+			return nil, nopIter, err
 		}
 		if !has {
 			s.invalidPreloadRequestsCount.Inc()
-			return nil, nil
+			return nil, nopIter, nil
 		}
-		if from.Add(-stalenessDelta).Before(last) && through.Add(stalenessDelta).After(first) {
```

Comment:
IIUC we now only handle staleness into the past, while we also handled it into the future here before. Is that an intended behavior change? That is, we only care about samples between now and 5 minutes into the past vs. 5 minutes into the past to 5 minutes into the future.